include/nbl/asset/interchange/IAssetLoader.h

```diff
@@ -301,6 +337,50 @@ class IAssetLoader : public virtual core::IReferenceCounted
+	{
+		return interm_getAssetInHierarchy(_mgr, _file, _supposedFilename, getFindParams(_params), _hierarchyLevel);
+	}
+	SAssetBundle interm_getAssetInHierarchy_find(IAssetManager* _mgr, const std::string& _filename, const IAssetLoader::SAssetLoadParams& _params, uint32_t _hierarchyLevel, IAssetLoader::IAssetLoaderOverride* _override)
+	{
+		return interm_getAssetInHierarchy(_mgr, _filename, getFindParams(_params), _hierarchyLevel, _override);
+	}
+	SAssetBundle interm_getAssetInHierarchy_find(IAssetManager* _mgr, io::IReadFile* _file, const std::string& _supposedFilename, const IAssetLoader::SAssetLoadParams& _params, uint32_t _hierarchyLevel)
+	{
+		return interm_getAssetInHierarchy(_mgr, _file, _supposedFilename, getFindParams(_params), _hierarchyLevel);
+	}
+	SAssetBundle interm_getAssetInHierarchy_find(IAssetManager* _mgr, const std::string& _filename, const IAssetLoader::SAssetLoadParams& _params, uint32_t _hierarchyLevel)
+	{
+		return interm_getAssetInHierarchy(_mgr, _filename, getFindParams(_params), _hierarchyLevel);
+	}
+	static IAssetLoader::SAssetLoadParams getReloadParams(IAssetLoader::SAssetLoadParams _params, uint32_t _restoreLevels, uint32_t _hierLevel)
+	{
+		_params.cacheFlags = ECF_DUPLICATE_UNTIL_LEVEL(_restoreLevels + _hierLevel);
+		return _params;
+	}
+	SAssetBundle interm_getAssetInHierarchy_reload(uint32_t _restoreLevels, IAssetManager* _mgr, io::IReadFile* _file, const std::string& _supposedFilename, const IAssetLoader::SAssetLoadParams& _params, uint32_t _hierarchyLevel, IAssetLoader::IAssetLoaderOverride* _override)
+	{
+		return interm_getAssetInHierarchy(_mgr, _file, _supposedFilename, getReloadParams(_params, _restoreLevels, _hierarchyLevel), _hierarchyLevel);
+	}
+	SAssetBundle interm_getAssetInHierarchy_reload(uint32_t _restoreLevels, IAssetManager* _mgr, const std::string& _filename, const IAssetLoader::SAssetLoadParams& _params, uint32_t _hierarchyLevel, IAssetLoader::IAssetLoaderOverride* _override)
+	{
+		return interm_getAssetInHierarchy(_mgr, _filename, getReloadParams(_params, _restoreLevels, _hierarchyLevel), _hierarchyLevel);
+	}
+	SAssetBundle interm_getAssetInHierarchy_reload(uint32_t _restoreLevels, IAssetManager* _mgr, io::IReadFile* _file, const std::string& _supposedFilename, const IAssetLoader::SAssetLoadParams& _params, uint32_t _hierarchyLevel)
+	{
+		return interm_getAssetInHierarchy(_mgr, _file, _supposedFilename, getReloadParams(_params, _restoreLevels, _hierarchyLevel), _hierarchyLevel);
+	}
+	SAssetBundle interm_getAssetInHierarchy_reload(uint32_t _restoreLevels, IAssetManager* _mgr, const std::string& _filename, const IAssetLoader::SAssetLoadParams& _params, uint32_t _hierarchyLevel)
+	{
+		return interm_getAssetInHierarchy(_mgr, _filename, getReloadParams(_params, _restoreLevels, _hierarchyLevel), _hierarchyLevel);
+	}
```

Comment:
could we do this via "perfect forwarding" or a similar template so I'm not seeing all this duplicate code

Reply:
i tried and not really, im ofc in a tricky way but wont be readable. Because of the need to modify load params

Reply:
Could something like this work?

```cpp
template<typename T>
T forwardButModify(T&& t, extra_parameter_t extra)
{
   if constexpr (std::is_same_v<T,TypeWeWantToModify>)
      return modifiedType(t); // should probably do separate versions for when the type is a value, const ref or rvalue ref
   return std::forward<T>(t);
}


template<typename... Args>
auto f_with_extra(extra_parameter_t extra, Args&&... args)
{
   return f(forwardButModify<Args>(args)...);
}
```

Reply:
yea, i've thought abut something very similar but decided those explicit overloads will be much more legible
if its ok for you, i can do something like this though

Reply:
but no, not this, forward is a cast to reference type, so this is not taht easy

Reply:
hmm fine, leave as is.. ist just painful to read.